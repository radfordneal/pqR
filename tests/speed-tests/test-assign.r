source("time.r")
 
test.name <- "assign.simple"

f <- test.cmp (function (n,a,b) 
{ for (i in 1:n) 
  { w <- a+1
    x <- a+b
    y <- a-b
    z <- a*b
  }
  w+x+y+z
})

sys.time(r <- f(1000000,1.1,2.2))
print(r)


test.name <- "assign.vecmat"

f <- test.cmp (function (n,j,v) 
{ vec <- (1:100)/100
  for (i in 1:n) 
  { vec[j] <- v
  }
  sum(vec)
})

sys.time(r <- f(1000000,17,1.1))
print(r)

sys.time(r <- f(1000000,15:17,1.1))
print(r)


f <- test.cmp (function (n,j,v) 
{ vec <- (1:100)/100
  names(vec) <- paste(c("a","b","c","d","e","f","g","h","i","j"),1:100,sep="")
  for (i in 1:n) 
  { vec[j] <- v
  }
  sum(vec)
})

sys.time(r <- f(1000000,"g17",1.1))
print(r)

sys.time(r <- f(1000000,c("e15","f16","g17"),1.1))
print(r)


f <- test.cmp (function (n,j,k,v) 
{ mat <- matrix ((1:100)/100, 10, 10)
  for (i in 1:n) 
  { mat[j,k] <- v
  }
  sum(mat)
})

sys.time(r <- f(1000000,7,2,1.1))
print(r)

sys.time(r <- f(1000000,5:7,2,1.1))
print(r)


test.name <- "assign.largevec"

f <- test.cmp (function (n,j,v) 
{ vec <- (0:10000)/10000
  for (i in 1:n) 
  { vec[j] <- v
  }
  sum(vec)
})

sys.time(r <- f(30000,201:8000,1.1))
print(r)

sys.time(r <- f(30000,201:8000,(1:100)*1.1))
print(r)

sys.time(r <- f(30000,201:8000,(201:8000)*1.1))
print(r)


test.name <- "assign.list"

f <- test.cmp (function (n) 
{ lis <- list(ab=3,cd=4,ef=5,gh=6)
  for (i in 1:n) 
  { lis$ef <- 10
  }
  lis
})

sys.time(r <- f(1000000))
print(unlist(r))


f <- test.cmp (function (n,j) 
{ lis <- list(ab=3,cd=4,ef=5,gh=6)
  for (i in 1:n) 
  { lis[[j]] <- 10
  }
  lis
})

sys.time(r <- f(1000000,"ef"))
print(unlist(r))

sys.time(r <- f(1000000,3))
print(unlist(r))


f <- test.cmp (function (n,j) 
{ lis <- list(ab=3,cd=4,ef=5,gh=6)
  for (i in 1:n) 
  { lis[j] <- 10
  }
  lis
})

sys.time(r <- f(1000000,"ef"))
print(unlist(r))

sys.time(r <- f(1000000,3))
print(unlist(r))

sys.time(r <- f(1000000,c("ef","ab")))
print(unlist(r))

sys.time(r <- f(1000000,c(3,1)))
print(unlist(r))


test.name <- "assign.complicated"

f <- test.cmp (function (n,j,k,v) 
{ lis <- list (a=1, b=(1:11)/11, c=list(xa=10,ya=20,xb=30,yb=40))
  for (i in 1:n) 
  { lis[[j]][k] <- v
  }
  lis
})

sys.time(r <- f(500000,2,3,5))
print(unlist(r))

sys.time(r <- f(500000,"b",3,5))
print(unlist(r))


f <- test.cmp (function (n,k,v) 
{ lis <- list (a=1, b=(1:11)/11, c=list(xa=10,ya=20,xb=30,yb=40))
  for (i in 1:n) 
  { lis$b[k] <- v
  }
  lis
})

sys.time(r <- f(500000,3,5))
print(unlist(r))

sys.time(r <- f(500000,c(3,1),5))
print(unlist(r))


f <- test.cmp (function (n,v) 
{ lis <- list (a=1, b=(1:11)/11, c=list(xa=10,ya=20,xb=30,yb=40))
  for (i in 1:n) 
  { lis$c$xb <- v
  }
  lis
})

sys.time(r <- f(500000,5))
print(unlist(r))


f <- test.cmp (function (n,v) 
{ lis <- list (a=1, b=matrix(4,2,2), c=10)
  for (i in 1:n) 
  { diag(lis$b)[2] <- 5
  }
  lis
})

sys.time(r <- f(60000,5))
print(unlist(r))


test.name <- "assign.named"

f <- test.cmp (function (n) 
{ u <- numeric(1000)
  for (i in 1:n)
  { v <- u
    v <- 0
    u[10] <- 1
  }
})

sys.time(f(500000))

f <- test.cmp (function (n) 
{ u <- numeric(1000)
  w <- 1
  for (i in 1:n)
  { v <- w
    v <- 0
    u[10] <- 1
  }
})

sys.time(f(500000))
