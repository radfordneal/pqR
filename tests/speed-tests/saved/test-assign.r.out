
---- assign.simple 

Time:
[1] 6.72

---- assign.vecmat 

Time:
[1] 51.43
Time:
[1] 53.32
Time:
[1] 51.43
Time:
[1] 53.32
Time:
[1] 51.43
Time:
[1] 53.32

---- assign.largevec 

Time:
[1] 10382.89
Time:
[1] 435092.9
Time:
[1] 35184093

---- assign.list 

Time:
ab cd ef gh 
 3  4 10  6 
Time:
ab cd ef gh 
 3  4 10  6 
Time:
ab cd ef gh 
 3  4 10  6 
Time:
ab cd ef gh 
 3  4 10  6 
Time:
ab cd ef gh 
 3  4 10  6 
Time:
ab cd ef gh 
10  4 10  6 
Time:
ab cd ef gh 
10  4 10  6 

---- assign.complicated 

Time:
          a          b1          b2          b3          b4          b5 
 1.00000000  0.09090909  0.18181818  5.00000000  0.36363636  0.45454545 
         b6          b7          b8          b9         b10         b11 
 0.54545455  0.63636364  0.72727273  0.81818182  0.90909091  1.00000000 
       c.xa        c.ya        c.xb        c.yb 
10.00000000 20.00000000 30.00000000 40.00000000 
Time:
          a          b1          b2          b3          b4          b5 
 1.00000000  0.09090909  0.18181818  5.00000000  0.36363636  0.45454545 
         b6          b7          b8          b9         b10         b11 
 0.54545455  0.63636364  0.72727273  0.81818182  0.90909091  1.00000000 
       c.xa        c.ya        c.xb        c.yb 
10.00000000 20.00000000 30.00000000 40.00000000 
Time:
          a          b1          b2          b3          b4          b5 
 1.00000000  0.09090909  0.18181818  5.00000000  0.36363636  0.45454545 
         b6          b7          b8          b9         b10         b11 
 0.54545455  0.63636364  0.72727273  0.81818182  0.90909091  1.00000000 
       c.xa        c.ya        c.xb        c.yb 
10.00000000 20.00000000 30.00000000 40.00000000 
Time:
         a         b1         b2         b3         b4         b5         b6 
 1.0000000  5.0000000  0.1818182  5.0000000  0.3636364  0.4545455  0.5454545 
        b7         b8         b9        b10        b11       c.xa       c.ya 
 0.6363636  0.7272727  0.8181818  0.9090909  1.0000000 10.0000000 20.0000000 
      c.xb       c.yb 
30.0000000 40.0000000 
Time:
          a          b1          b2          b3          b4          b5 
 1.00000000  0.09090909  0.18181818  0.27272727  0.36363636  0.45454545 
         b6          b7          b8          b9         b10         b11 
 0.54545455  0.63636364  0.72727273  0.81818182  0.90909091  1.00000000 
       c.xa        c.ya        c.xb        c.yb 
10.00000000 20.00000000  5.00000000 40.00000000 
Time:
 a b1 b2 b3 b4  c 
 1  4  4  4  5 10 

---- assign.named 

Time:
Time:
